content/blog/real-time-smart-city-traffic-monitoring-using-microservices-based-streaming-architecture-part-2.md

```diff
@@ -0,0 +1,331 @@
+
+* A program to feed sensor data into MapR Event Store/Kafka. This part will be using fake data modeled by a vehicle simulation coded with Log Synth. We’ll use the MapR <a target='\_blank'  href='http://docs.confluent.io/2.0.0/kafka-rest/docs/index.html'>Kafka-rest</a> proxy implementation (just introduced with MEP 2.0)  to add the data with Python.
+* A JVM-language application that will read data from the stream and pass it to a KieSession. The minimal code to get this working is surprisingly small.
+
+To edit the rules, we deploy the Workbench on Wildfly 10, which is a fairly straightforward process. <a target='\_blank'  href='http://betzelblog.blogspot.jp/2015/02/setting-up-drools-workbench-and.html'>Check this blog post</a> for instructions, or read the Drools Documentation. Installing Wildfly is pretty simple; <a target='\_blank'  href='http://developer-should-know.com/post/112230363742/how-to-install-wildfly-as-a-service-on-linux'>see this blog</a> for great instructions on how to install it as a service on Centos/RHEL (it’s for Wildfly, but the same instructions work for 9 and 10).
+
+We made a single configuration change to Wildfly. We changed the port to 28080 instead of 8080, as it is already used by the sandbox. Wildfly runs in standalone mode, so the configuration file is in WILDFLY_HOME/standalone/configuration/standalone.xml.
+
+For monitoring, we let the streaming architecture work for us. We use the <a target='\_blank'  href='https://streamsets.com/products/dataops-platform/data-collector-engine/'>Streamset Data collector</a> to easily redirect the sensor data to Elasticsearch so that we can actually monitor the system with a nice dashboard with Kibana. To set up Streamsets with MapR Event Store requires some work with version 1.6, or from the <a target='\_blank'  href='https://streamsets.com/documentation/datacollector/latest/help/#Install_Config/MapR-Prerequisites.html'>official Streamsets documentation</a>).
+
+Finally, installation and setup of Elasticsearch and Kibana is <a target='\_blank'  href='https://www.digitalocean.com/community/tutorials/how-to-install-elasticsearch-logstash-and-kibana-elk-stack-on-centos-7'>well documented on Centos/RHEL</a>.
+
+For production, all those parts can be readily separated to run on separate servers. They can be run on either cluster nodes or edge nodes. If it’s a MapR cluster, installing the MapR Client and pointing it to the cluster CLDB nodes will be all the configuration needed for full access to the streams. For an Apache Kafka cluster, refer to the <a target='\_blank'  href='http://kafka.apache.org/documentation'>official Kafka documentation</a>.
+
+## Traffic Monitoring Prototype - How To
+
+**Creating the Streams with maprcli**
+
+The first task is to create streams and topics for our application. To do this, it’s a best practice to create a volume for streams first. As user mapr, type from the command line:
```

Comment:
Shouldn't this read "As a MapR user, type from..."?

![image](https://user-images.githubusercontent.com/77496261/137516146-1ca4be0e-e846-49c2-aaec-d5b6b8ca1944.png)

Reply:
i rewrote it. "mapr" is the user name, so "As a MapR user" doesn't quite work here.